### PR TITLE
Removed Guava (left in tests), bump Gradle and all deps. (this is 2nd part of https://github.com/java-json-tools/jackson-coreutils/pull/3)

### PR DESCRIPTION
**Pull request by soberich**
Hi,
We really want to remove Guava as it is on of the largest deps in war though required by this dependency (and jackson-coreutils). This would ease the choice to depend or not on Guava.
Please kindly review.
__N.B. NOW BOTH  `jackson-coreutils` and `json-patch` could be left without Guava.__
__See https://github.com/java-json-tools/jackson-coreutils/pull/3__
No "optimizations" made. Just a plain migration with practically coping and pasting logic from Guava's methods (Guava is a great library but we just have a JAX-RS web-app and Guava is a too big contributor to classpath).

Overview.1. Removed Guava (left in tests)
2. Bump gradlew to 5.0 and adjust scripts for compatibility
3. Builds and tests run now on Java 6-11
4. Bump all versions to top available.
EDIT: @huggsboson @fge just to ping you guys.

### CODE REVIEW

**Comment by soberich** [comment]
CI failed due to version 1.10-(SNAPSHOT) of `jackson-coreutils` is not available in maven. I am sure that is clear. It may run fine with whatever version.

**Comment by soberich** [comment]
@huggsboson @fge hey guys, please, I see a lot of things simply hang on ` java-json-tools` repos. Just to keep pace with things around maybe it's worth to transfer ownership / propose other solutions / accept pull-request , something like that? So many projects transitively depend on this one and others, and it enforces some constraints which were fine for sometime, but time passes by and maybe it is worth to think about such things and take some action. (please take me correct and forgive it I have expressed myself in weird way, English is not Java, you know, no compile time checks))
I really appreciate a response. 
Best Regards.

**Comment by mmannerm** [comment]
It would be really great if we can get Guava out from this library. https://nvd.nist.gov/vuln/detail/CVE-2018-10237

**Comment by soberich** [comment]
@mmannerm I have no idea why these repos are so dead. 
https://github.com/java-json-tools/json-patch
https://github.com/java-json-tools/jackson-coreutils
https://github.com/json-path/JsonPath

completely dead. 
Funny that thousands of projects use them.
Need some action.

**Comment by Capstan** [comment]
I've been fiddling today with json-patch to bring it into modernity, so if you are interested in pursuing this, we can do that. I'm going to release it at least once with modern Guava to avoid the CVEs.

**Comment by soberich** [comment]
@Capstan I have asked before, but didn't fully get the answer. The whole idea behind was to **remove** Guava completely from (as many as possible) projects as it is totally non-needed. I felt we are on the same page here simply because why not to remove the biggest unnecessary dependency when you easily can?? I see in many repos/discussions in recent time this slowly boiled down to [at least] updating Guava. Boiling this down to updating Guava neglect the point of work and discussion. The benefits are clear and I see not reason to go conservative here.  Can we get back and stick to "remove Guava" as the goal? Or maybe I interpreted incorrectly those comment/PR by you?

**Comment by Capstan** [comment]
@soberich I answered previously at https://github.com/java-json-tools/jackson-coreutils/pull/3#issuecomment-526613603: Where there are CVEs, the easiest route forward is to update the dependencies to avoid the CVEs, even if it maintains a dependency on Guava, especially since removing Guava is likely to break forward compatibility. Updating to a modern Guava to allow people to sync and be safe does not prohibit removing it in a follow-on commit.

File: src/main/java/com/github/fge/jsonpatch/diff/DiffProcessor.java [hunk 1]
```diff
@@ -26,12 +26,9 @@
 import com.github.fge.jsonpatch.JsonPatchOperation;
 import com.google.common.base.Equivalence;
```
Comment (Capstan) [inline src/main/java/com/github/fge/jsonpatch/diff/DiffProcessor.java]: So `Equivalence` and `Predicate` still need to be removed. I've patched and synched through and I still get these errors. I'll take few minutes to see if they're easily removable.

File: src/main/java/com/github/fge/jsonpatch/diff/DiffProcessor.java [hunk 2]
```diff
@@ -26,12 +26,9 @@
...
 import com.google.common.base.Predicate;
-import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.Lists;
 
 import javax.annotation.Nullable;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (Capstan) [inline src/main/java/com/github/fge/jsonpatch/diff/DiffProcessor.java]: I've exploded these into individual imports for clarity.

File: src/main/java/com/github/fge/jsonpatch/diff/DiffProcessor.java (hunk 1 above)
Comment (Capstan) [reply src/main/java/com/github/fge/jsonpatch/diff/DiffProcessor.java]: Done. See #66

**Comment by Capstan** [comment]
Rolled the above into #66

**Pull request closed without merging**